app/features/worksheet/tools.py

```diff
@@ -0,0 +1,311 @@
+from typing import List, Tuple, Dict, Any
+
+import os
+
+from langchain_google_vertexai import VertexAI
```

Comment:
Should use the class, `from langchain_google_genai import GoogleGenerativeAI` 

Use class, can utilize the API key from aistudio Google

Reply:
Fixed it.